src/ORM/Association/HasMany.php

```diff
@@ -134,6 +181,67 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     }
 
     /**
+     * Deletes/sets null the related objects according to the dependency between source and targets and foreign key nullability
+     * Skips deleting records present in $remainingEntities
+     *
+     * @param array $properties array of foreignKey properties
+     * @param EntityInterface $entity the entity which should have its associated entities unassigned
+     * @param Table $target The associated table
+     * @param array $remainingEntities Entities that should not be deleted
+     * @return void
+     */
+    protected function _unlinkAssociated(array $properties, EntityInterface $entity, Table $target, array $remainingEntities)
```

Comment:
In the current status, if you add new records, they will be saved, but the old associated values will be replaced.
You should keep in mind that this is only happening if you define a save strategy as replace when associating the tables.